odxtools/ecuvariantmatcher.py

```diff
@@ -1,171 +1,32 @@
 # SPDX-License-Identifier: MIT
-from enum import Enum
-from typing import Dict, Generator, List, Optional
+from typing import List
 
-from .diaglayers.diaglayer import DiagLayer
-from .diaglayers.diaglayertype import DiagLayerType
 from .diaglayers.ecuvariant import EcuVariant
-from .diagservice import DiagService
-from .exceptions import OdxError, odxassert, odxrequire
-from .matchingparameter import MatchingParameter
-from .odxtypes import ParameterValue
-from .response import Response
+from .variantmatcher import VariantMatcher
 
 
-class EcuVariantMatcher:
-    """EcuVariantMatcher implements the matching algorithm of ecu variants according to their
-    ECU-VARIANT-PATTERN according to ISO 22901-1.
+class EcuVariantMatcher(VariantMatcher):
```

Comment:
Not sure whether this class is still needed, since all behavior is defined in the base class. The docstring is useful, though.

Reply:
I agree with @dmholtz you don't need a logicless wrapper for that

Reply:
Where is the `isinstance` check that you are referring to?

Reply:
currently there's none, but if you want, I can modify the test... (as said below: this is exactly the same situation as with `ComplexDop`). Also, I think it is more intuitive to use `EcuVariantMatcher` if you want to do ECU variant matching...

Reply:
so the class exists to cover a case that doesn't exist?

Reply:
ok, I budge about the subclasses issue: https://github.com/mercedes-benz/odxtools/pull/394/commits/d0b7481b26f64f0a172bd1a53e940202eacf191c .

for accessing the matching parameters, I'll stay firm on using a method (`.get_matching_parameters()`) instead of a property (`.matching_variant_parameters`) because reading the former IMO has less potential for confusion because nobody will try to look this up in the XML data...